src/data/D2ApiIndicator.ts

```diff
@@ -223,11 +231,46 @@ export class D2ApiIndicator {
                     disaggregation: undefined,
                     initialDisaggregation: undefined,
                     categories: [],
+                    suggestedCompanions: suggestedCompanions,
                 });
             })
             .value();
     }
 
+    private getSuggestedCompanionCodes(
+        attributes: D2Config["attributes"],
+        attributeValues: D2Attribute[]
+    ): SuggestedIndicator[] {
+        const mandatoryIndicatorsValues = attributeValues.find(
+            attribute => attribute.attribute.id === attributes.mandatoryCompanionIndicator.id
+        );
+
+        const optionalIndicatorsValues = attributeValues.find(
+            attribute => attribute.attribute.id === attributes.optionalCompanionIndicator.id
+        );
+
+        const mandatoryCodes = this.getCompanionCodesFromAttribute(mandatoryIndicatorsValues);
+        const optionalCodes = this.getCompanionCodesFromAttribute(optionalIndicatorsValues);
+
+        const mandatoryIndicators: Maybe<SuggestedIndicator> =
+            mandatoryCodes.length > 0 ? { codes: mandatoryCodes, type: "mandatory" } : undefined;
+
+        const optionalIndicators: Maybe<SuggestedIndicator> =
+            optionalCodes.length > 0 ? { codes: optionalCodes, type: "optional" } : undefined;
+
+        return _([mandatoryIndicators, optionalIndicators]).compact().value();
+    }
+
+    private getCompanionCodesFromAttribute(attributeValue: Maybe<D2Attribute>): Code[] {
+        if (!attributeValue) return [];
+
+        const mandatoryCodes = attributeValue?.value.split(this.companionSeparator);
+
+        return _(mandatoryCodes)
+            .compactMap(code => code.trim())
```

Comment:
Beware that est-lodash->compact (thus also compactMap), only compacts null and undefined. Your usage here suggests you want to remove empty string codes. If that's the case you add an extra `.filter(...)`